app/hotels/src/allHotels/__tests__/HotelsPaginationContainer.test.js

```diff
@@ -1,63 +1,96 @@
 // @flow
 
+import * as React from 'react';
+import renderer from 'react-test-renderer';
 import { Logger } from '@kiwicom/mobile-shared';
 
 import { HotelsPaginationContainer } from '../HotelsPaginationContainer';
+import { HotelsContext } from '../../HotelsContext';
+
+jest.mock('../RenderSearchResults.js');
+jest.mock('../../filter/FilterStripe');
 
 const defaultProps = (stats: Object = { maxPrice: 100, minPrice: 50 }) => ({
-  setCurrentSearchStats: jest.fn(),
   data: {
     allAvailableBookingComHotels: {
       stats,
     },
   },
+  relay: {
+    hasMore: () => false,
+  },
 });
 
+const renderComponent = (
+  props: Object = {},
+  setCurrentSearchStats: Function = () => {},
+) =>
+  renderer.create(
+    <HotelsContext.Provider
+      // $FlowExpectedError: Just passing what is needed for the test
+      value={{
+        // $FlowExpectedError: Just passing what is needed for the test
+        actions: {
+          setCurrentSearchStats,
+        },
+      }}
+    >
+      <HotelsPaginationContainer {...props} />
+    </HotelsContext.Provider>,
+  );
+
 it('calls Logger when mounted', () => {
   const spy = jest.spyOn(Logger, 'ancillaryDisplayed');
-  // $FlowExpectedError: Passing only props needed for this test
-  const Component = new HotelsPaginationContainer(defaultProps());
-  Component.componentDidMount();
+  const wrapper = renderComponent(defaultProps());
+  wrapper.update(); // useEffect is not called on intial render, but when browser is idle, https://github.com/facebook/react/issues/14050#issuecomment-447888631
```

Comment:
It's a shame that hooks make the tests slightly less readable and understandable: why do you need to call `update` twice?

Reply:
Ah okay, to verify the logger was called only at the initial render.

Reply:
I don't need to call update twice. I call update twice, in order to see that the Logger is called only once.